apps/judicial-system/web/src/utils/utils.spec.ts

```diff
@@ -449,4 +450,85 @@ describe('Step helper', () => {
       expect(ind).toEqual(false)
     })
   })
+
+  describe('removeTabs', () => {
+    test('should replace a single tab with a single space', () => {
+      // Arrange
+      const str = '\t'
+
+      // Act
+      const res = replaceTabs(str)
+
+      // Assert
+      expect(res).toEqual(' ')
+    })
+
+    test('should replace multiple consecutive tabs with a single space', () => {
+      // Arrange
+      const str = '\t\t\t'
+
+      // Act
+      const res = replaceTabs(str)
+
+      // Assert
+      expect(res).toEqual(' ')
+    })
+
+    test('should remove multiple consecutive tabs with a leading space', () => {
+      // Arrange
+      const str = ' \t\t\t'
+
+      // Act
+      const res = replaceTabs(str)
+
+      // Assert
+      expect(res).toEqual(' ')
+    })
+
+    test('should remove multiple consecutive tabs with a trailing space', () => {
+      // Arrange
+      const str = '\t\t\t '
+
+      // Act
+      const res = replaceTabs(str)
+
+      // Assert
+      expect(res).toEqual(' ')
+    })
+
+    test('should process a complicated string with tabs', () => {
+      // Arrange
+      const str =
+        'Lorem\t ipsum dolor \t\tsit amet,\t\t\t\tconsectetur \t\t\t adipiscing elit.'
+
+      // Act
+      const res = replaceTabs(str)
+
+      // Assert
+      expect(res).toEqual(
+        'Lorem ipsum dolor sit amet, consectetur  adipiscing elit.',
+      )
+    })
+
+    test('should handle undefined', () => {
```

Comment:
skv typescript definition-inu á þessi function ekki að taka in undifined. þannig mér finnst þetta test ekki meika sense, nema að við látum functionið leyfa undifined.

Reply:
Ég er sammála þér að því leyti að typescript skilgreiningin leyfir okkur ekki að skrifa kóða þar sem það er augljóst á compile time að undefined gildi er sent inn í functionina. Hins vegar enforce-ar typescript ekki strict typing á runtime svo argumentið getur orðið undefined. Við reynum að verja okkur fyrir þessu í clientinum með því að setja gildi í tóma strenginn áður en util function-irnar okkar eru notaðar, en það munu poppa upp holur í þeim vörnum og þá munum við lenda í null reference villu sem krassar clientinum. Vörnin gegn því er að verja okkur fyrir null reference í util function-inni sjálfri og ég nota til þess ? og bý svo til test til að sannreyna að það sé að virka. Til að þetta make-i meira sense gætum við sagt að functionin megi taka inn undefined en ættum við þá ekki að gera það sama við allar function-ir sem taka inn strengi eða önnur gildi og vinna á þeim? Þessi gildi gætu öll orðið undefined og/eða null á runtime og það er alltaf góð regla að verja sig fyrir svoleiðis. Betra að nota belti og axlabönd en að forritið krassi.

Reply:
held að það ætti ekki að getað komið inn undifined, nema að við séum að typa eitthvað vitluast eða það sé villa í vafranum. 

En þetta er ágætis nálgun svo sem, að vera viss um að þetta myndi virka án þess að menga kóðann með óþarfa undifined týpum.

Reply:
Sammála þessu. Ég er einmitt að hugsa um villur í vafranum þar sem undefined er sent inn. Þetta hefur einmitt gerst oftar en einu sinni. Við stoppum í götin en með nýjum kóða læðast inn nýjar villur.